### PR TITLE
DAOS-8055 utils: Separate daos_perf and vos_perf

### PR DESCRIPTION
**Pull request by NiuYawei**
To get rid of server module dependency from daos_perf, the original
daos_perf is split into two binaries:

- daos_perf : for original 'daos' and 'echo' modes only.
- vos_perf  : for original 'vos' mode.

Some obsolete options are removed, more command help messages are
added.

dts lib is also modified to get rid of dependency on vos, and
daos_racer is now independent of vos as well.

Signed-off-by: Niu Yawei <yawei.niu@intel.com>
Co-authored-by: Jeff Olivier <jeffrey.v.olivier@intel.com>

### CODE REVIEW

**Comment by NiuYawei** [comment]
CentOS 8 build failed for "fatal: unable to access 'http://github.com/willemt/CLinkedListQueue/': Failed to connect to github.com port 80: Connection timed out"

**Review by wangdi1 — DISMISSED** [review]

File: debian/changelog [hunk 1]
```diff
@@ -1,4 +1,10 @@
-daos (1.3.104-2) unstable; urgency=medium
+daos (1.3.104-4) unstable; urgency=medium
+  [ Niu Yawei ]
+  * Add vos_perf
+
+ -- Niu Yawei <yawei.niu@intel.com>  Thu, 05 Aug 2021 14:22:00 +0800
+
+daos (1.3.104-3) unstable; urgency=medium
```
Comment (jolivier23) [inline debian/changelog:7]: I think it's fine to have a different release number for debian

**Review by jolivier23 — APPROVED** [review]
You've lost a race again on spec.   But otherwise, looks good to me

**Review by jolivier23 — APPROVED** [review]

**Review by wangdi1 — DISMISSED** [review]

**Review by jolivier23 — CHANGES_REQUESTED** [review]
I just tried this locally and noticed that readelf -d daos_perf shows a dependence on libdaos_common_pmem.so.   I uploaded #6427 which creates a separate client/server list so vos_perf depends on libdaos_common_pmem.so but  daos_perf only depends on libdaos_common.so

**Review by jolivier23 — APPROVED** [review]

**Comment by jolivier23** [comment]
I took the liberty of merging those changes to move this along while you're out.

**Comment by jolivier23** [comment]
Johann did some documentation updates for daos_perf -t VOS so that will need to be updated

**Comment by NiuYawei** [comment]
ping reviewers, could you review this?

**Review by jolivier23 — APPROVED** [review]

**Review by wangdi1 — APPROVED** [review]

**Pull request merged**